### DISCUSSION TITLE
comments not loading

### DISCUSSION THREAD

**Discussion by RavikumarTulugu** [umputun/remark42#1210] [Q&A] [answered]
I know this is another silly issue, but pls bear with me. i have done 99% of debugging before posting this. 
I am trying to load comments from a subdomain.  I am using openlitespeed webserver with virtual host for each subdomain.
I am using context type as proxy with external server [ remark : localhost: 8080 ]. The remark server is hosted behind openlitespeed server on port 8080 with out ssl. I am using openlitespeed as a reverse proxy with rewrite rules https to http. 

on tcp dump  i am seeing following requests and responses to and from remark server.
HTTP: GET /comments/web/last-comments.mjs HTTP/1.1
HTTP: HTTP/1.1 404 Not Found

is the URI path reaching the remark server correct ?? 

here are my rewrite rules. 
RewriteEngine On
RewriteCond %{HTTPS} =on
RewriteCond %{REQUEST_URI}  =/comments/*
RewriteRule  (.*) http://%{SERVER_NAME}:8080/$1 [R,L]

on chrome console : 
(index):37 GET https://www.playme.one/comments/web/last-comments.mjs net::ERR_ABORTED 404

I am not running docker and just running binary as below .  is there any setup needed before running the binary ??  
./remark42.linux-amd64 server --url=https://www.playme.one/comments --secret="abcdefghijkl" --site="foobillard" --dbg

please advise.

**Comment by umputun** [comment]
moved to the discussion as this seems to be "how to set up" sort of question

**Comment by umputun** [comment]
from what I can see, remark42 is reachable on  https://www.playme.one/comments, i.e. https://www.playme.one/comments/ping hits it. However, I don't see the expected response on https://www.playme.one/comments/web, which is strange. Smth is not right with your configuration, not sure what

**Comment by RavikumarTulugu** [comment]
Please find my frontend configuration. 

<script>
  var remark_config = {
    host: 'https://www.playme.one/comments', 
    site_id: 'foobillard',
    components: ['last-comments'], 
    max_shown_comments: 20,
    theme: 'dark', 
    locale: 'en', 
    show_email_subscription: false, 
    simple_view: false 
  };
</script>
<script>
!function(e,n){for(var o=0;o<e.length;o++){var r=n.createElement("script"),c=".js",d=n.head||n.body;"noModule"in r?(r.type="module",c=".mjs"):r.async=!0,r.defer=!0,r.src=remark_config.host+"/web/"+e[o]+c,d.appendChild(r)}}(remark_config.components||["embed"],document);
</script>

**Comment by RavikumarTulugu** [comment]
pasting http request body hitting the remark42 server, i suspect the rewrite rules messing up the URL which remark42 is rejecting. 
     GET /comments/web/last-comments.mjs HTTP/1.1
        host: www.playme.one
        user-agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:95.0) Gecko/20100101 Firefox/95.0
        accept: */*
        accept-language: en-US,en;q=0.5
        accept-encoding: gzip             
        origin: https://foobillard.playme.one
        referer: https://foobillard.playme.one/

   HTTP/1.1 404 Not Found
        Access-Control-Allow-Credentials: true
        Access-Control-Allow-Origin: *
        Access-Control-Expose-Headers: Authorization
        App-Name: remark42
        App-Version: v1.8.1-82f8f6c9-20210504T18:25:06
        Author: umputun
        Content-Type: text/plain; charset=utf-8
        Vary: Origin
        X-Content-Type-Options: nosniff
        Date: Fri, 31 Dec 2021 15:56:00 GMT
        Content-Length: 19

**Comment by umputun** [comment]
I don't think this is your proxy issue, but rather a limitation of our binary distribution, which has a very limited support for adjusting asset's URL on the startup. We have a ticket about this and planning to change the way how we do it - instead of modifying multiple html/js files initially, to always serve them with the internal router.

In a mean time I'd suggest switching to the recommended docker-based configuration.

**Comment by RavikumarTulugu** [comment]
Thank you for getting back, the reason i opted for binary is to keep my setup and arch simple. i can wait for any patch or new binary release.

**Comment by RavikumarTulugu** [comment, marked as answer]
as suspected rewrite rules were messing up. after fixing rewrite rules, every thing worked fine.

**Reply by dayeggpi** [reply]
can you please share the adjusted rewrite rules please to make this work ? I am also trying to make remark42 work with openlitespeed and I don't know how to do the reverse proxy part...

**Reply by paskal** [reply]
Here is the link which might help you: https://www.dasmirnov.net/blog/remark42-on-apache-without-a-subdomain/.

Also, I think this issue might be helpful to you @hycday: https://github.com/umputun/remark42/issues/759

**Comment by RavikumarTulugu** [comment]
RewriteEngine On
RewriteCond %{HTTPS} on
RewriteCond %{REQUEST_URI}  /comments/(.*) 
RewriteRule  /comments/(.*)  http://remark42/$1 [P]
